CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -35,6 +35,9 @@ end
 
 function Staff:tickDay()
   Humanoid.tickDay(self)
+  if self.fired or self.dead then
+    return
+  end
```

Comment:
The function comment on Humanoid:tickDay states that the result of it (returns boolean) should determine the processing of the caller, ie if not Humanoid.tickDay(self) then return false and same in Doctor with Staff class. Staff:tickDay doesn't return boolean (nor does Doctor but its result is not used anyway).

Patient has that logic, where it does waiting logic then does the Humand.tickDay which allows it to abort out of Patient:tickDay based on that result. As you already added self.fired to Humanoid:tickDay then it could serve the same purpose but Staff:tickDay would then need to return true/false for Doctor class.